pkg/util/dynamichelper/dynamichelper.go

```diff
@@ -30,26 +36,35 @@ import (
 	_ "github.com/Azure/ARO-RP/pkg/util/scheme"
 )
 
+// Interface defines a set of methods which allow retrieving and updating
+// GVKs from the kubernets api.
 type Interface interface {
-	Refresh() error
-	EnsureDeleted(ctx context.Context, groupKind, namespace, name string) error
-	Ensure(ctx context.Context, objs ...kruntime.Object) error
+	Refresh() error                                                                              // Refresh the in-memory storage of the GVK
+	GetResource(ctx context.Context, groupKind, namespace, name string) (kruntime.Object, error) // Retrieve GVK resource
+	EnsureDeleted(ctx context.Context, groupKind, namespace, name string) error                  // Ensures the deletion of the GVK
+	Ensure(ctx context.Context, objs ...kruntime.Object) error                                   // Ensures the GVK exists and is correct
 }
 
-type dynamicHelper struct {
-	GVRResolver
+// DynamicHelper is a wrapper around a kubernetes admin api that provides utility functions
+// for retrieving any GVK with generic types.
+type DynamicHelper struct {
```

Comment:
- [ ] Check to make sure that we should be exposing the dynamic helper struct itself outside of the package... this is what we have New functions for.